### ISSUE TITLE
command to switch servers

### ISSUE THREAD

**Issue by AnthonyFuller** [Grover-c13/MultiAdmin#151]
I want to type stuff into server1 console but server2 console is the active one I need a command like

SWITCH [SERVER NAME/FOLDER]
switches the console

**Comment by AnthonyFuller** [comment]
BUMP

**Comment by Dankrushen** [comment]
I doubt this is very possible, I assume you are using SSH, so I'd suggest making multiple instances (If you're using Linux, I'd suggest something like `screen`) and running MultiAdmin on each one. I don't believe keeping track of multiple consoles like that is very possible, and if it is, then not easy at all.

**Comment by PatPeter** [comment]
@Grover-c13 I'm thinking of using my MultiAdmin fork to try implementing this sometime this week. The way it will work is the admin types:

`select <folder name>`

And each command fired after that will be fired into that server instance. In addition, the logs will look as follows:

```[01:48:00] [<folder name>] Accepted authentication token of...```

So that users can differentiate which server each line of output is coming from.

**Comment by Grover-c13** [comment]
Sounds good, I'm sure a lot of people would like that feature.

**Comment by PatPeter** [comment]
I hate how I've been so swamped with fixing my plugins that I haven't had time to work on this, but trust me I think of this every time I have to input a command in one of my five servers at random hoping it goes to the right one.

**Comment by PatPeter** [comment]
I'm lowkey angry that I haven't had the time to fully work on this while trying to resolve my own server's issues.

**Comment by AnthonyFuller** [comment]
Any progress?

**Comment by PatPeter** [comment]
Whenever I get back to #168 to fix this new annoying bug, I'll definitely work on fixing #151 as well. Right now, I'm so swamped with work for my 7-8 servers that I don't know when I'll be getting to MultiAdmin development.

**Comment by PatPeter** [comment]
If my Patreon plugin wasn't broken all this weekend and during the New Year, I would have been working on this issue instead. Kept trying to tell one of my servers to restart and it was restarting the first instance instead of the second.

**Comment by Dankrushen** [comment]
Sadly I didn't end up making this yet, my new plan is to simply output all the server's messages to console (potentially have the command to filter to just a specific server and back?), but have the Server ID displayed with the message and log to their own files like how it would work normally.

**Comment by PatPeter** [comment]
Yeah I've been doing research on and off on how we would do it. The problem is any libraries that I would use are either for Linux on Linux or Windows on Windows. I was thinking of MultiAdmin starting a Linux screen for each server, or splitting the output into two divided columns. Posting the output from all servers is a step in the right direction though.

I can't think of how to do this in Mono without storing all the output to a buffer and risking a memory leak. We might have to do a "select" command to select which server to send commands to instead of a "switch" command to change the view along with the destination of the commands.

> 
> 
> @Grover-c13 I'm thinking of using my MultiAdmin fork to try implementing this sometime this week. The way it will work is the admin types:
> 
> `select <folder name>`
> 
> And each command fired after that will be fired into that server instance. In addition, the logs will look as follows:
> 
> `[01:48:00] [<folder name>] Accepted authentication token of...`
> 
> So that users can differentiate which server each line of output is coming from.
> 
> I don't want to name it switch since it isn't switching the console, just changing which server receives the input.

**Comment by storm37000** [comment]
im doing this with something i made called MegaAdmin even though its sort of hacky, but it works well so thats all that matters really.
